src/arbitraries/MappedArbitrary.ts

```diff
@@ -32,8 +39,17 @@ export class MappedArbitrary<A, B> extends Arbitrary<B> {
   }
 
   canGenerate(pick: FluentPick<B>) {
-    /* && pick.value === this.f(pick.original) */
-    return this.baseArbitrary.canGenerate({value: pick.original, original: pick.original})
+    const inverseValues = !(this.config && this.config.inverseFunction) ? [pick.original] :
+      (Array.isArray(this.config.inverseFunction(pick.value)) ?
+        this.config.inverseFunction(pick.value) as A[] :
+        [this.config.inverseFunction(pick.value)])
+
+    for (const value of inverseValues) {
+      if (this.baseArbitrary.canGenerate({value, original: pick.original}))
+        return true
+    }
+
+    return false
```

Comment:
`return inverseValues.some(value => this.baseArbitrary.canGenerate({value, original: pick.original}))` ?

Reply:
Nice suggestion! Already considered it.